### PR TITLE
added the changes in appmgt/store module to carbon-store

### PR DESCRIPTION
**Pull request by lakshani**

### CODE REVIEW

File: jaggery-modules/store/scripts/user.js [hunk 1]
```diff
@@ -7,18 +7,18 @@ var user = {};
...
-    var cleanUsername = function (username) {
+    user.cleanUsername = function (username) {
         /**
          * this is a one-way hash function, @ is replaced if the user name is an email
          * if the user name is coming from a secondery user store it will be second.com/user hence
          * "/" will be replaced.
          */
 
-        return username.replace('@', ':').replace('/', ':');
+        return username.replace('@', '-AT-').replace('/', ':');
```
Comment (splinter) [inline jaggery-modules/store/scripts/user.js]: I believe some Identity components have issues with -AT- (uppercase AT)

File: jaggery-modules/store/scripts/server.js [hunk 2]
```diff
@@ -131,8 +148,11 @@ var server = {};
                 secured: false
             };
         }
-        //loads the tenant if it hasn't been loaded
-        server.loadTenant(obj);
+
+        if(obj.tenantId != -1){
```
Comment (splinter) [inline jaggery-modules/store/scripts/server.js]: Shall we change the magic number -1 to a constant such as INVALID_TENANTID?

File: jaggery-modules/store/scripts/server.js [hunk 3]
```diff
@@ -103,25 +103,42 @@ var server = {};
...
      */
     server.tenant = function (request, session) {
-        var obj, domain, user,
+        var obj, domain, user, urlDomain,
             carbon = require('carbon');
-        /*matcher = new URIMatcher(request.getRequestURI());
-         if (matcher.match('/{context}/' + opts.tenantPrefix + '/{domain}') ||
-         matcher.match('/{context}/' + opts.tenantPrefix + '/{domain}/{+any}')) {
-         domain = matcher.elements().domain; */
+        matcher = new URIMatcher(request.getRequestURI());
```
Comment (splinter) [inline jaggery-modules/store/scripts/server.js]: Can you make this line:

var matcher = new URIMatcher(request.getRequestURI());

*Note:* This is an issue which was not introduced by this PR but rather a long standing problem

File: jaggery-modules/store/scripts/user.js [hunk 4]
```diff
@@ -7,18 +7,18 @@ var user = {};
 
     var USER_ROLE_PREFIX = 'Internal/private_';
 
-    var cleanUsername = function (username) {
+    user.cleanUsername = function (username) {
```
Comment (splinter) [inline jaggery-modules/store/scripts/user.js]: Are we using the cleanUsername function anywhere else?

File: jaggery-modules/store/scripts/user.js (hunk 1 above)
Comment (lakshani) [inline jaggery-modules/store/scripts/user.js]: We test this js (with -AT-) with identity 5.0.9-SNAPSHOT, kernal-4.4.5 and registry -4.5.3 versions without any issue.

**Pull request merged**